src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -0,0 +1,442 @@
+      0.1,
+      "Precursor MZ tolerance used during the annotation phase.\n"
+      "For each transition in the target list, annotateSpectra() looks for "
+      "the first spectrum whose precursor MZ is close enough (+-mz_tolerance_) "
+      "to the transition's MZ.\n"
+      "Also the spectrum's precursor RT is checked against the transition RT."
+    );
+
+    params.setValue("mz_unit_is_Da", "true", "Unit to use for mz_tolerance_ and fwhm_threshold_: true for Da, false for ppm.");
+    params.setValidStrings("mz_unit_is_Da", ListUtils::create<String>("false,true"));
+
+    params.setValue("sgolay_frame_length", 15, "The number of subsequent data points used for smoothing.\nThis number has to be uneven. If it is not, 1 will be added.");
+    params.setValue("sgolay_polynomial_order", 3, "Order of the polynomial that is fitted.");
+    params.setValue("gauss_width", 0.2, "Gaussian width in Da or ppm, estimated peak size.");
+    params.setValue("use_gauss", "true", "Use Gaussian filter for smoothing (alternative is Savitzky-Golay filter)");
+    params.setValidStrings("use_gauss", ListUtils::create<String>("false,true"));
+    params.setValue("signal_to_noise", 1.0, "Signal-to-noise threshold at which a peak will not be extended any more. Note that setting this too high (e.g. 1.0) can lead to peaks whose flanks are not fully captured.");
+    params.setMinFloat("signal_to_noise", 0.0);
+
+    params.setValue("peak_height_min", 0.0, "Used in pickSpectrum(), a peak's intensity needs to be >= peak_height_min_ for it to be picked.");
+    params.setMinFloat("peak_height_min", 0.0);
+    params.setValue("peak_height_max", 4e6, "Used in pickSpectrum(), a peak's intensity needs to be <= peak_height_max_ for it to be picked.");
+    params.setMinFloat("peak_height_max", 0.0);
+    params.setValue("fwhm_threshold", 0.0, "Used in pickSpectrum(), a peak's FWHM needs to be >= fwhm_threshold_ for it to be picked.");
+    params.setMinFloat("fwhm_threshold", 0.0);
+
+    params.setValue("tic_weight", 1.0, "TIC weight when scoring spectra.");
+    params.setMinFloat("tic_weight", 0.0);
+    params.setValue("fwhm_weight", 1.0, "FWHM weight when scoring spectra.");
+    params.setMinFloat("fwhm_weight", 0.0);
+    params.setValue("snr_weight", 1.0, "SNR weight when scoring spectra.");
+    params.setMinFloat("snr_weight", 0.0);
+  }
+
+  void TargetedSpectraExtractor::annotateSpectra(
+    const std::vector<MSSpectrum>& spectra,
+    const TargetedExperiment& targeted_exp,
+    std::vector<MSSpectrum>& annotated_spectra,
+    FeatureMap& features
+  )
+  {
+    annotated_spectra.clear();
+    features.clear(true);
+    const std::vector<ReactionMonitoringTransition> transitions = targeted_exp.getTransitions();
+    for (UInt i=0; i<spectra.size(); ++i)
+    {
+      MSSpectrum spectrum = spectra[i];
+      const double divisor = (String)param_.getValue("rt_unit") == "seconds" ? 1.0 : 60.0;
+      const double spectrum_rt = spectrum.getRT() / divisor;
```

Comment:
I am not really sure whether that is correct, I `spectrum.RT()` is always in seconds, see
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/KERNEL/MSSpectrum.h#L172